cns/wireserver/net.go

```diff
@@ -49,6 +54,49 @@ func GetPrimaryInterfaceFromResult(res *GetInterfacesResult) (*InterfaceInfo, er
 	return nil, ErrNoPrimaryInterface
 }
 
+// Gets secondary interface details for swiftv2 secondary nics scenario
+func GetSecondaryInterfaceFromResult(res *GetInterfacesResult, macAddress string) (*InterfaceInfo, error) {
+	for _, i := range res.Interface {
+		// skip if primary
+		if i.IsPrimary {
+			continue
+		}
+
+		// skip if no subnets
+		if len(i.IPSubnet) == 0 {
+			continue
+		}
+
+		logger.Printf("i.MacAddress is %s", strings.Split(i.MacAddress, ":"))
+		newMacAddress := regexp.MustCompile(`[^a-zA-Z0-9 ]+`).ReplaceAllString(macAddress, "")
+		if i.MacAddress == strings.ToUpper(newMacAddress) {
+			// get the second subnet
+			s := i.IPSubnet[0]
+			gw, err := calculateGatewayIP(s.Prefix)
+			if err != nil {
+				return nil, err
+			}
+
+			secondaryIP := ""
+			for _, ip := range s.IPAddress {
+				if !ip.IsPrimary {
+					secondaryIP = ip.Address
+				}
+			}
+			secondaryIPs := []string{}
+			secondaryIPs = append(secondaryIPs, secondaryIP)
+
+			return &InterfaceInfo{
+				Subnet:       s.Prefix,
+				IsPrimary:    false,
+				Gateway:      gw.String(),
+				SecondaryIPs: secondaryIPs,
+			}, nil
+		}
+	}
+	return nil, ErrNoSecondaryInterfaces
```

Comment:
since we will get only one secondary interface at a time as we discussed, so let's rename ErrNoSecondaryInterfaces to ErrNoSecondaryInterface